pkg/virt-operator/resource/apply/instancetypes.go

```diff
@@ -23,19 +23,29 @@ func (r *Reconciler) createOrUpdateInstancetypes() error {
 	return nil
 }
 
-func (r *Reconciler) createOrUpdateInstancetype(instancetype *instancetypev1beta1.VirtualMachineClusterInstancetype) error {
-	instancetypeClient := r.clientset.VirtualMachineClusterInstancetype()
-
-	foundObj, err := instancetypeClient.Get(context.Background(), instancetype.Name, metav1.GetOptions{})
-	if err != nil && !errors.IsNotFound(err) {
-		return err
+func (r *Reconciler) findInstancetype(name string) (*instancetypev1beta1.VirtualMachineClusterInstancetype, error) {
+	obj, exists, err := r.stores.ClusterInstancetype.GetByKey(name)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, errors.NewNotFound(v1.Resource("VirtualMachineClusterInstancetype"), name)
+	}
+	foundObj, ok := obj.(*instancetypev1beta1.VirtualMachineClusterInstancetype)
+	if !ok {
+		return nil, fmt.Errorf("unknown object within VirtualMachineClusterInstancetype store")
 	}
+	return foundObj, nil
+}
+
+func (r *Reconciler) createOrUpdateInstancetype(instancetype *instancetypev1beta1.VirtualMachineClusterInstancetype) error {
+	foundObj, err := r.findInstancetype(instancetype.Name)
```

Comment:
We should cover the case where the returned error is `not nil && is different from IsNotFound`

Reply:
Added a check for this case.

Reply:
I'd have preferred previous style:
```
if err != nil && !errors.IsNotFound(err) {
		return err
```
:)